### PR TITLE
(#127) Cleanup instructions. Revamp unit-tests that are failing. Integrate to CI build.yml

### PR DESCRIPTION
**Pull request by gapisback**
This commit cleans-up the md-formatting in a majority of docs. Some bug-fixes done in few unit-test programs, which are now enabled as part of CI: 

Mainly around `certifier_service/certlib/`
Cleanup `certlib/instructions.md`; Add test_data/generate_policy.sh to CI.

---
NOTE: To reviewer: this change-set is mainly cleaning up md-file formatting.

Discovered and fixed stray bugs in a couple of unit-tests, and enabled them in CI, so we are protected against regressions in the future.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -36,15 +36,74 @@ jobs:
...
+        cd ../certprotos
+        protoc --go_opt=paths=source_relative --go_out=. --go_opt=Mcertifier.proto= ./certifier.proto
+
+        echo " "
+        echo "---- Running certlib/go test ... ----"
+        echo " "
+        #! Run certlib/ Go unit-tests, which use policy_key_file.bin policy_cert_file.bin
+        #! from test_data/ dir.
+        cd ../certlib
+        go test
```
Comment (gapisback) [inline .github/workflows/build.yml:103]: I found this `go test` was not compiling. Fixed a couple of bugs along the way and rearranged the test-code to run off of `test_data` subdir.

Above steps are required in order to run this test.

Enabling this test in CI protects us against regressions ...

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -36,15 +36,74 @@ jobs:
...
+        ../../../utilities/cert_utility.exe                    \
+            --operation=generate-policy-key-and-test-keys      \
+            --policy_key_output_file=policy_key_file.bin       \
+            --policy_cert_output_file=policy_cert_file.bin     \
+            --platform_key_output_file=platform_key_file.bin   \
+            --attest_key_output_file=attest_key_file.bin
+        set +x
+
+        set -x
+        ./generate_policy.sh
```
Comment (gapisback) [inline .github/workflows/build.yml:77]: Add this test to CI ... to mainly ensure that the documented instructions still continue to work correctly as doc'ed.

File: Doc/Suggested_features.md [hunk 3]
```diff
@@ -1,22 +1,22 @@
-Suggestions for upgrade
+# Enhancements and Feature Requests
```
Comment (gapisback) [inline Doc/Suggested_features.md:1]: Much of the change set from here-on, below, is simply cleaning up md-files format, to make the docs a bit more readable & useful.

File: certifier_service/certlib/cert1_test.go [hunk 4]
```diff
@@ -799,7 +799,7 @@ func TestArtifacts(t *testing.T) {
 		t.Error("Can't parse parent Certificate")
 	}
 
-	cert := ProduceAdmissionCert(privateIssuerKey, newParentCert, subjKey, "testSubject", "",
+	cert := ProduceAdmissionCert("", privateIssuerKey, newParentCert, subjKey, "testSubject", "",
```
Comment (gapisback) [inline certifier_service/certlib/cert1_test.go:802]: Minor bug fix.

File: certifier_service/certlib/cert1_test.go [hunk 5]
```diff
@@ -113,8 +113,8 @@ func TestDominance(t *testing.T) {
 func TestKeys(t *testing.T) {
 	fmt.Print("\nTestKeys\n")
 
-	keyFile := "policy_key_file.bin"
-	certFile := "policy_cert_file.bin"
+	keyFile := "./test_data/policy_key_file.bin"
+	certFile := "./test_data/policy_cert_file.bin"
```
Comment (gapisback) [inline certifier_service/certlib/cert1_test.go:117]: Change facilitates enabling this test thru CI.

**Review by yelvmw — APPROVED** [review]
LGTM. I think the added tests make sense. Thank you!

**Pull request merged**